SingularityService/src/main/java/com/hubspot/singularity/resources/DeployResource.java

```diff
@@ -87,25 +87,31 @@ public SingularityRequestParent deploy(@ApiParam(required=true) SingularityDeplo
     final String requestId = checkNotNullBadRequest(deploy.getRequestId(), "DeployRequest must have a non-null requestId");
 
     SingularityRequestWithState requestWithState = fetchRequestWithState(requestId);
-    SingularityRequest request = requestWithState.getRequest();
 
     authorizationHelper.checkForAuthorization(requestWithState.getRequest(), user, SingularityAuthorizationScope.WRITE);
 
+    SingularityRequest newRequestData = requestWithState.getRequest();
+    boolean hasUpdatedRequest = false;
```

Comment:
i think you can short-circuit this to `final boolean hasUpdatedRequest = deployRequest.getNewRequestData().isPresent()`, since an invalid updated request will cause an exception